src/pages/StocktakeEditPage.js

```diff
@@ -141,7 +140,7 @@ export class StocktakeEditPage extends React.Component {
   assignReason = stocktakeItem => {
     const { database } = this.props;
     if (stocktakeItem.shouldApplyReason) {
-      this.onOpenReasonModal();
+      this.setState({ isReasonsModalOpen: true, currentStocktakeItem: stocktakeItem });
```

Comment:
I think a comment here would be a good idea, keep a paper trail.

Reply:
YOU'RE A COMMENT

Reply:
I actually don't know what you'd have me comment here.

`// Open reason selection modal`?

Reply:
A comment on the `if` statement (as explained more below). The code reads: 

```
IF shouldApplyReason
THEN
  updateState
ELSE
  applyReasonToBatches
```

I think this is confusing without a comment.

```